### PR TITLE
Reorganized standalone & example config files into hocon

### PR DESCRIPTION
**Pull request by marcinczeczko**
## Description
After migrating to HOCON form, a default configuration file for whole Knot.x is required. The config should have mentioned all configuration parameters with a comments describing its purpose.

## Motivation and Context
Existing config is missleading and user need to consult multiple places to find possible options to configure

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the [code style](https://github.com/Cognifide/knotx/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. Documentation to be covered in #396 

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: knotx-example/knotx-example-app/config/application.conf [hunk 1]
```diff
@@ -9,63 +10,107 @@ modules = [
...
 
-repo {
-  domain = localhost
-  port = 3001
-}
+########### Globals ###########
+global {
+  serverPort = 8092
+
+  snippetTagName = script
```
Comment (tomaszmichalak) [inline knotx-example/knotx-example-app/config/application.conf]: Maybe it is the time to introduce the custom tag name like "knotx:snippet" :) ?

File: knotx-standalone/config/application.conf [hunk 2]
```diff
@@ -1,12 +1,103 @@
...
-    "serviceAdapter=io.knotx.adapter.service.http.HttpServiceAdapterVerticle"
-  ]
+  "server=io.knotx.server.KnotxServerVerticle"
+  "httpRepo=io.knotx.repository.http.HttpRepositoryConnectorVerticle"
+  "splitter=io.knotx.splitter.FragmentSplitterVerticle"
+  "assembler=io.knotx.assembler.FragmentAssemblerVerticle"
+  "hbsKnot=io.knotx.knot.templating.HandlebarsKnotVerticle"
+  "serviceKnot=io.knotx.knot.service.ServiceKnotVerticle"
+  # Enable action knot if you want to have support for form submittions. Follow the actionKnot configuration to set it up
+  # "actionKnot=io.knotx.knot.action.ActionKnotVerticle"
```
Comment (tomaszmichalak) [inline knotx-standalone/config/application.conf]: Action Knot is disabled but configured in the routing.

File: knotx-example/knotx-example-app/config/application.conf (hunk 1 above)
Comment (Skejven) [reply knotx-example/knotx-example-app/config/application.conf]: We definitely have to change the default to e.g. `knotx:snippet`. But not yet, let's not break backward compability.

Comment (marcinczeczko) [reply knotx-example/knotx-example-app/config/application.conf]: I agree with skejven. Making it default to something else, not only break the backward compatibility, but also introduces a lot of refactors of tests - that anyway need to be rebuilt, changed, reorganized...

File: knotx-standalone/config/application.conf (hunk 2 above)
Comment (marcinczeczko) [reply knotx-standalone/config/application.conf]: Yes, I'm thinking actually about enabling actionKnot and leaving the config as it's now - and clarify in the config that if you're not using any module, just comment it out or remove.

File: knotx-example/knotx-example-app/config/includes/serviceKnot.conf [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Event bus address of the handlebars Knot
+address = ${global.address.knot.service}
+
+# An
```
Comment (Skejven) [inline knotx-example/knotx-example-app/config/includes/serviceKnot.conf]: Some missing comment here?

File: knotx-example/knotx-example-app/config/includes/serviceAdapter.conf [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Event bus address of the Basic HTTP Service Adapter
```
Comment (Skejven) [inline knotx-example/knotx-example-app/config/includes/serviceAdapter.conf]: Do we need here an example (commented out) how to setup connection via SSL?

Comment (marcinczeczko) [reply knotx-example/knotx-example-app/config/includes/serviceAdapter.conf]: Added more stuff

**Pull request merged**